src/dmd/cppmangle.d

```diff
@@ -995,8 +995,19 @@ private final class CppMangleVisitor : Visitor
         // fake mangling for fields to fix https://issues.dlang.org/show_bug.cgi?id=16525
         if (!(d.storage_class & (STC.extern_ | STC.field | STC.gshared)))
         {
-            d.error("Internal Compiler Error: C++ static non-`__gshared` non-`extern` variables not supported");
-            fatal();
+            if (global.params.targetOS == TargetOS.OSX)
+            {
+                // OSX emits TLS wrapper functions, shouldn't be hard to do.
+                // These start with _ZTW
+                d.error("C++ `thread_local` is unsupported on OSX");
+                fatal();
+            }
+            else if (global.params.cplusplus == CppStdRevision.cpp98)
+            {
+                d.error("C++ `thread_local` variables are not supported in C++98 compatibiity mode");
```

Comment:
C++-98 supports `__thread`, or is that a gnu extension?

Reply:
it is an extension, also support by clang.